Kernel/Bus/VirtIO/Device.cpp

```diff
@@ -115,9 +115,19 @@ UNMAP_AFTER_INIT void Device::initialize()
             cfg->offset = capability.read32(0x8);
             cfg->length = capability.read32(0xc);
             dbgln_if(VIRTIO_DEBUG, "{}: Found configuration {}, bar: {}, offset: {}, length: {}", VirtIO::determine_device_class(address), (u32)cfg->cfg_type, cfg->bar, cfg->offset, cfg->length);
-            if (cfg->cfg_type == ConfigurationType::Common)
+            if (cfg->cfg_type == ConfigurationType::Common) {
```

Comment:
Why are we only mapping in common configs? All configs can have bars associated with them

Reply:
I'm not 100% sure about this, the thing that trips me up is that accessing the bar was guarded by `m_use_mmio`, which is only set if we have a cfg of type ConfigurationType::Common. If there are no Common ones, we get an assertion failed if we try to access the mapping for the bar later.

Reply:
Right, but if we *do* have a common one we're going to have to setup the bars for all of them. It might be nicer to just iterate over the configs after we found all of them (if m_use_mmio was set)

Reply:
Fair enough, I'll push another revision.